src/baal/utils/metrics.py

```diff
@@ -207,11 +210,12 @@ def update(self, output=None, target=None):
         # this is to make sure handling 1.0 value confidence to be assigned to a bin
         output = np.clip(output, 0, 0.9999)
 
-        for cls in range(self.n_cls):
-            for pred, t in zip(output[:, cls], target):
-                bin_id = int(math.floor(pred * self.n_bins))
-                self.samples[cls, bin_id] += 1
-                self.tp[cls, bin_id] += int(cls == t)
+        for pred, t in zip(output, target):
+            t = int(t)  # Force the conversion
+            conf, p_cls = pred.max(), pred.argmax()
+            bin_id = int(math.floor(conf * self.n_bins))
+            self.samples[t, bin_id] += 1
+            self.tp[t, bin_id] += int(p_cls == t)
```

Comment:
if one of the classes does not appear in prediction results, we don't calculate ece for? It should be 0, no?

Reply:
Well if a class appears in target and we never predict it, the accuracy for it would be 0 and the error would be according to that.

Reply:
right its taken care of. cool